File: app/src/main/java/life/plank/juna/zone/view/adapter/common/NotificationAdapter.java
package life.plank.juna.zone.view.adapter.common;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import java.util.ArrayList;

import butterknife.BindView;
import butterknife.ButterKnife;
import life.plank.juna.zone.R;

public class NotificationAdapter extends RecyclerView.Adapter<NotificationAdapter.NotificationViewHolder> {

    private static final String TAG = NotificationAdapter.class.getSimpleName();
    private Context context;
    private ArrayList<String> notificationList;


    public NotificationAdapter(Context context, ArrayList<String> notificationList) {
        this.context = context;
        this.notificationList = notificationList;
    }

    @Override
    public NotificationViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new NotificationViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.item_notification, parent, false));

    }

    @Override
    public void onBindViewHolder(NotificationViewHolder holder, int position) {
        holder.notification.setText(notificationList.get(position));
    }

    @Override
    public int getItemCount() {
        return notificationList.size();
    }

    public class NotificationViewHolder extends RecyclerView.ViewHolder {

        @BindView(R.id.profile_pic)
        ImageView profileImage;
        @BindView(R.id.notification)
        TextView notification;
        @BindView(R.id.time)
        TextView time;

        NotificationViewHolder(View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }
    }
}

File: app/src/main/java/life/plank/juna/zone/view/adapter/common/NotificationAdapter.kt
package life.plank.juna.zone.view.adapter.common

import android.view.*
import androidx.recyclerview.widget.RecyclerView
import kotlinx.android.synthetic.main.item_notification.view.*
import life.plank.juna.zone.R
import life.plank.juna.zone.data.model.notification.SocialNotification
import life.plank.juna.zone.data.network.interfaces.RestApi
import life.plank.juna.zone.util.common.execute
import life.plank.juna.zone.util.facilis.onDebouncingClick
import life.plank.juna.zone.util.sharedpreference.PreferenceManager.Auth.getToken

class NotificationAdapter(private val restApi: RestApi) : RecyclerView.Adapter<NotificationAdapter.NotificationViewHolder>() {

    val notificationList: MutableList<SocialNotification> = ArrayList()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): NotificationViewHolder {
        return NotificationViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.item_notification, parent, false))
    }

    override fun onBindViewHolder(holder: NotificationViewHolder, position: Int) {
        notificationList[position].run {
            holder.itemView.run {
                notification_message.text = notificationMessage
                notification_time.setText(R.string.now)
            }
            holder.itemView.onDebouncingClick {
                restApi.setNotificationAsRead(id, getToken()).execute()
//            TODO: open specified notification
            }
        }
    }

    override fun getItemCount(): Int = notificationList.size

    fun update(notificationList: MutableList<SocialNotification>) {
        this.notificationList.clear()
        this.notificationList.addAll(notificationList)
        notifyDataSetChanged()
    }

    fun clear() {
        notificationList.clear()
        notifyDataSetChanged()
    }

    class NotificationViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView)
}